### PR TITLE
First pass features for data acquisition microservice

### PR DESCRIPTION
**Pull request by acron0**
The first feature for this microservice is the 'request to share' feature which manages the state for the process of a user requesting data from another user.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,36 +1,24 @@
...
-### That You Think
-### Might be Useful
+``` bash
+curl -G http://localhost:8086/query --data-urlencode "q=CREATE DATABASE metrics
+```
 
 ## License
 
-Copyright © 2016 FIXME
+Copyright © 2016 MastodonC
```
Comment (otfrom) [inline README.md]: Mastodon C Ltd

File: src/kixi/data_acquisition/db/inmemory.clj [hunk 2]
```diff
@@ -0,0 +1,40 @@
+(ns kixi.data-acquisition.db.inmemory
```
Comment (otfrom) [inline src/kixi/data_acquisition/db/inmemory.clj:1]: these in memory stubs always worry me as we often aren't doing the same shape of data storage when we use c\* or others.

File: src/kixi/data_acquisition/metrics.clj [hunk 3]
```diff
@@ -0,0 +1,142 @@
+(ns kixi.data-acquisition.metrics
```
Comment (otfrom) [inline src/kixi/data_acquisition/metrics.clj:1]: this feels to me like something we'd factor out into a library. wdyt?

File: src/kixi/data_acquisition/request_to_share.clj [hunk 4]
```diff
@@ -0,0 +1,107 @@
+(ns kixi.data-acquisition.request-to-share
+  (:require [kixi.data-acquisition.datastore :refer :all :as ds]
+            [kixi.data-acquisition.db :refer :all :as db]
```
Comment (otfrom) [inline src/kixi/data_acquisition/request_to_share.clj]: why the refer?

File: src/metrics/reporters/influxdb.clj [hunk 5]
```diff
@@ -0,0 +1,55 @@
+(ns metrics.reporters.influxdb
```
Comment (otfrom) [inline src/metrics/reporters/influxdb.clj:1]: this feels like a library too

File: test/kixi/data_acquisition/unit/db/inmemory_test.clj [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+         (component/system-map
+          :db (map->DbInMemory {})))]
+    (reset! db (:db system))
+    (all-tests)
+    (component/stop-system system)
+    (reset! db nil)))
+
+(use-fixtures :each create-db)
+
+(deftest create-table-no-index
```
Comment (otfrom) [inline test/kixi/data_acquisition/unit/db/inmemory_test.clj:22]: most of the tests in here look like they could be described as fdef generative tests. Might be worth doing that for future iterations.

**Review by otfrom — CHANGES_REQUESTED** [review]
Only necessary change is the copyright notice.

File: src/kixi/data_acquisition/db/inmemory.clj (hunk 2 above)
Comment (acron0) [reply src/kixi/data_acquisition/db/inmemory.clj:1]: I try and use them as a way to define the interface I want to use. For example, I _shouldn't care_ how the DB implementation re-formats my table names as long as I can continue to refer to them in an idiomatic way (e.g. `:table-name` would need to be `"table_name"` in C\* but is fine in InMemory version)

File: src/kixi/data_acquisition/metrics.clj (hunk 3 above)
Comment (acron0) [reply src/kixi/data_acquisition/metrics.clj:1]: Agree but not a priority

File: src/kixi/data_acquisition/request_to_share.clj (hunk 4 above)
Comment (acron0) [reply src/kixi/data_acquisition/request_to_share.clj]: Premature, will remove.

File: src/metrics/reporters/influxdb.clj (hunk 5 above)
Comment (acron0) [reply src/metrics/reporters/influxdb.clj:1]: Agree, as above.

File: test/kixi/data_acquisition/unit/db/inmemory_test.clj (hunk 6 above)
Comment (acron0) [reply test/kixi/data_acquisition/unit/db/inmemory_test.clj:22]: Yeah, I have no experience of this but would be interested to learn when we have the time

File: README.md (hunk 1 above)
Comment (acron0) [reply README.md]: Will fix

File: src/kixi/data_acquisition/metrics.clj (hunk 3 above)
Comment (otfrom) [reply src/kixi/data_acquisition/metrics.clj:1]: I agree. More one to think about as soon we will do this a 3rd time.

File: src/kixi/data_acquisition/db/inmemory.clj (hunk 2 above)
Comment (otfrom) [reply src/kixi/data_acquisition/db/inmemory.clj:1]: I more worry about how the structure of something in c\* or es can affect how you would model your data and that we might make something that either didn't fit or didn't take advantage of it. I think we are OK for now, but it is a question I think we should ask ourselves when using mocks like this.

File: test/kixi/data_acquisition/unit/db/inmemory_test.clj (hunk 6 above)
Comment (otfrom) [reply test/kixi/data_acquisition/unit/db/inmemory_test.clj:22]: OK. We can all look at it together at some point.

**Pull request merged**